flintrock/ec2.py

```diff
@@ -57,33 +58,47 @@ def __init__(
             vpc_id: str,
             master_instance: 'boto3.resources.factory.ec2.Instance',
             slave_instances: "List[boto3.resources.factory.ec2.Instance]",
+            use_private_vpc,
```

Comment:
Perhaps then we need to add `subnet_id` as a cluster attribute and use that. Currently, the user can explicitly provide a subnet to use, or Flintrock will assume it can be determined automatically from the VPC, because a VPC can have a default subnet.

I don't think we need to query the master instance.